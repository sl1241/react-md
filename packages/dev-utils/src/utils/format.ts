import log from "loglevel"
import type { BuiltInParserName } from "prettier"
import prettier from "prettier"

function getParser(
    code: string,
    parser: BuiltInParserName | undefined,
    filePath: string | undefined
): BuiltInParserName | undefined {
    if (typeof parser === "string") {
        return parser
    }

    if (typeof filePath === "string") {
        return undefined
    }

    if (code.startsWith("{") || code.startsWith("[")) {
        return "json"
    }

    if (/^(@import|\.[a-z])/m.test(code) || /^\s+@(mixin|include)/.test(code)) {
        return "scss"
    }

    return "typescript"
}

export function format(
    code: string,
    parser?: BuiltInParserName,
    filepath?: FunctionMaybe<Nullable<string>>
): string {
    const resolvedParser = getParser(code, parser, filepath)
    try {
        return prettier.format(code, {
            filepath,
            parser: resolvedParser,
            proseWrap: resolvedParser === "markdown" ? "always" : undefined,
            singleQuote: resolvedParser === "scss" || resolvedParser === "babel",
        })
    } catch (e) {
        log.error(e)
        process.exit(1)
    }
}
