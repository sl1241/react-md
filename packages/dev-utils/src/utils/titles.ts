import { upperFirst } from "lodash"

/**
 * A simple utility that converts some string into a page/section title. This
 * really just capitalizes each word in the string after it has been split by
 * hyphens or capital letters.
 *
 * @param s - The string to convert to a title.
 * @param joinWith - A string that should be used to join all the parts together
 * to create a title.
 * @param capitals - Boolean if the string should be split on capital letters
 * instead of hyphens.
 * @returns A title-ized string.
 */
export function toTitle(s: string, joinWith = " ", capitals = false): string {
    if (/autocomplete/i.test(s)) {
        return "AutoComplete"
    }

    if (/^api$/i.test(s)) {
        return "API"
    }

    if (/sassdoc/i.test(s)) {
        return "SassDoc"
    }

    if (s === "[id]") {
        // TODO: Fix to use query params instead
        return "Layout"
    }

    return s
        .split(capitals ? /(?=[A-Z])/ : "-")
        .map((part) => {
            if (/^((v\d+)|(to))$/.test(part)) {
                return part
            }

            if (/^api$/i.test(part)) {
                return "API"
            }

            if (/^cdn$/i.test(part)) {
                return "CDN"
            }

            return upperFirst(part)
        })
        .join(joinWith)
}

/**
 * Converts a pathname into a breadcrumb type of title to display in the
 * documentation site.
 *
 * @param pathname - The current pathname to create a title for
 * @param statusCode - An optional status code to check against for custom error
 * cases.
 * @returns a string to display in the documentation site's breadcrumbs.
 */
export function toBreadcrumbPageTitle(
    pathname: string,
    statusCode?: FunctionMaybe<Nullable<number>>
): string {
    let title = ""
    if (statusCode) {
        switch (statusCode) {
            case 404:
                title = "Not Found"
                break
            default:
                title = "Server error"
        }
    } else if (/v\d+-to-v\d+$/.test(pathname)) {
        const [migration] = pathname.split("/").reverse()
        title = `Migration Guides - ${migration.replace(/-/g, " ")}`
    } else {
        const parts = pathname.split("/").filter((p) => !!p && !/packages/.test(p))
        title = parts.map((p) => toTitle(p)).join(" - ")
    }

    return `react-md${title ? ` - ${title}` : ""}`
}

/**
 * This is a utility function that will create a unique id for a title string.
 * The title string should normally be somewhere betwen 5-20 characters.
 *
 * This is generally used to create ids for links within the documentation page
 * for headers to generate quick links and table of contents.
 *
 * @param title - The title to create an id for.
 * @returns an id based on the title
 */
export function toId(title: string): string {
    return title
        .replace(/\/|\\|\[|]|-/g, "")
        .replace(/SCSS/g, "scss ")
        .replace(/CSS/g, "css ")
        .replace(/API/g, "api")
        .replace(/\s+/g, " ")
        .trim()
        .split(/\s|(?=[A-Z])/)
        .join("-")
        .toLowerCase()
}
