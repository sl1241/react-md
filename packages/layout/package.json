{
  "name": "@react-md/layout",
  "version": "5.1.4",
  "description": "This package is used to configure most of react-md as well as create some pre-defined layouts",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./types/index.d.ts",
  "sideEffects": [
    "dist/**/*"
  ],
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md.git",
    "directory": "packages/layout"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "homepage": "https://react-md.dev/packages/layout/demos",
  "keywords": [
    "accessibility",
    "a11y",
    "react-md",
    "material design",
    "react",
    "layout",
    "configuration",
    "context providers",
    "setup",
    "component",
    "components"
  ],
  "license": "MIT",
  "dependencies": {
    "@react-md/app-bar": "workspace:../app-bar",
    "@react-md/dialog": "workspace:../dialog",
    "@react-md/divider": "workspace:../divider",
    "@react-md/form": "workspace:../form",
    "@react-md/icon": "workspace:../icon",
    "@react-md/link": "workspace:../link",
    "@react-md/list": "workspace:../list",
    "@react-md/menu": "workspace:../menu",
    "@react-md/sheet": "workspace:../sheet",
    "@react-md/states": "workspace:../states",
    "@react-md/theme": "workspace:../theme",
    "@react-md/tooltip": "workspace:../tooltip",
    "@react-md/transition": "workspace:../transition",
    "@react-md/tree": "workspace:../tree",
    "@react-md/typography": "workspace:../typography",
    "@react-md/utils": "workspace:../utils",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@types/react": "^18.0.29",
    "csstype": "^3.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.9.0"
  },
  "peerDependencies": {
    "react": ">= 16.14",
    "react-dom": ">= 16.14"
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": false
    },
    "react-dom": {
      "optional": false
    }
  },
  "publishConfig": {
    "access": "public"
  }
}
