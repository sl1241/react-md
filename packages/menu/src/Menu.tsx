;
import { List } from "@react-md/list"
import { ConditionalPortal } from "@react-md/portal"
import { useScaleTransition } from "@react-md/transition"
import type { LabelA11y, LabelRequiredForA11y } from "@react-md/utils"

import { MenuKeyboardFocusProvider } from "./MenuKeyboardFocusProvider"
import { MenuWidget } from "./MenuWidget"
import type { MenuProps } from "./types"

/**
 * This component conditionally renders the {@link MenuWidget} with the
 * {@link MenuKeyboardFocusProvider} based on the `visible` prop with a CSS
 * transition that can be configured.
 *
 * @remarks \@since 5.0.0
 */
export const Menu = (
  {
    className,
    listStyle,
    listClassName,
    listProps,
    visible,
    temporary = true,
    horizontal = false,
    portal = temporary,
    portalInto,
    portalIntoId,
    appear,
    enter,
    exit,
    onEnter,
    onEntering,
    onEntered,
    onExit,
    onExiting,
    onExited,
    timeout,
    classNames,
    children,
    ref,
    ...props
  }: LabelRequiredForA11y<MenuProps & LabelA11y>
) => {
  const { elementProps, rendered, stage } = useScaleTransition({
    nodeRef: ref as any,
    className,
    transitionIn: visible,
    vertical: !horizontal,
    temporary,
    timeout,
    classNames,
    appear,
    enter,
    exit,
    onEnter,
    onEntering,
    onEntered,
    onExit,
    onExiting,
    onExited,
  })

  return (
    <ConditionalPortal
      portal={portal}
      portalInto={portalInto}
      portalIntoId={portalIntoId}
    >
      {rendered && (
        <MenuKeyboardFocusProvider horizontal={horizontal}>
          {/* @ts-ignore */}
          <MenuWidget
            hidden={!temporary && stage === "exited"}
            {...props}
            {...elementProps}
            horizontal={horizontal}
          >
            {/* @ts-ignore */}
            <List
              {...listProps}
              //@ts-ignore
              style={listStyle ?? listProps?.style}
              className={listClassName ?? listProps?.className}
              horizontal={horizontal}
              onClick={(event) => {
                //@ts-ignore
                listProps?.onClick?.(event)

                // this makes it so you can click on the menu/list without
                // closing the menu
                if (event.target === event.currentTarget) {
                  event.stopPropagation()
                }
              }}
            >
              {children}
            </List>
          </MenuWidget>
        </MenuKeyboardFocusProvider>
      )}
    </ConditionalPortal>
  )
}
