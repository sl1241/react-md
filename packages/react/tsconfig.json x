{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
