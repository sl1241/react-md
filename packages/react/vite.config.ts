import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
