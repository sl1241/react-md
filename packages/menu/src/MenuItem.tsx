

import { ListItem } from "@react-md/list"
import { useKeyboardFocusableElement } from "@react-md/utils"

import type { MenuItemProps } from "./types"

/**
 * This component is used as an "action" within a `Menu`/`DropdownMenu` that
 * implements some keyboard focus behavior. This component should generally have
 * an `onClick` event handler.
 *
 * @remarks \@since 5.0.0
 */
export const MenuItem = ({ className, children, role = "menuitem", tabIndex = -1, ref: nodeRef, ...props }: MenuItemProps) => {
  const ref = useKeyboardFocusableElement(nodeRef as any)
  return (
    <ListItem
      {...props}
      ref={ref}
      //@ts-ignore
      role={role}
      tabIndex={tabIndex}
      className={["rmd-menu-item", className]}
    >
      {children}
    </ListItem>
  )
}

