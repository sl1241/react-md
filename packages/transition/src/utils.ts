import { getViewportSize } from "@react-md/utils"
import type {
    CSSTransitionClassNames,
    CSSTransitionClassNamesObject,
    TransitionActions,
    TransitionTimeout,
    TransitionTimeoutObject,
} from "./types"
import { CSSProperties, $$ } from "voby"

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export interface TransitionTimeoutOptions extends Required<TransitionActions> {
    timeout: FunctionMaybe<TransitionTimeout>
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export function getTransitionTimeout({ timeout, appear, enter, exit, }: Readonly<TransitionTimeoutOptions>): Readonly<Required<TransitionTimeoutObject>> {
    const to = $$(timeout)

    if (typeof to === "number") {
        return {
            appear: appear ? to : 0,
            enter: enter ? to : 0,
            exit: exit ? to : 0,
        }
    }

    return {
        appear: (appear && (to.appear ?? to.enter)) || 0,
        enter: (enter && to.enter) || 0,
        exit: (exit && to.exit) || 0,
    }
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export interface CollapseSizing {
    maxHeight?: CSSProperties['maxHeight']
    paddingTop?: CSSProperties['paddingTop']
    paddingBottom?: CSSProperties['paddingBottom']
}

/**
 * A small util that will find the max-height, padding-top, and padding-bottom
 * for the provided element. This is really used to be able to transition the
 * max-height value since `max-height: auto` does not transition. The only way
 * to get transition is to change max-height values manually.
 *
 * @remarks \@since 4.0.0
 * @internal
 */
export function getElementSizing(element: HTMLElement | null): CollapseSizing {
    let maxHeight
    let paddingTop
    let paddingBottom
    if (element) {
        maxHeight = element.scrollHeight

        // clone the element (not deep) just to figure out it's padding without the
        // inline styles applied
        const cloned = element.cloneNode(false) as HTMLElement
        cloned.style.maxHeight = ""
        cloned.style.padding = ""
        cloned.style.paddingLeft = element.style.paddingLeft
        cloned.style.paddingRight = element.style.paddingRight
        cloned.style.visibility = "hidden"

        const container = element.parentElement || document.body
        container.appendChild(cloned)
        const style = window.getComputedStyle(cloned)
        if (style.paddingTop) {
            paddingTop = parseFloat(style.paddingTop)
        }

        if (style.paddingBottom) {
            paddingBottom = parseFloat(style.paddingBottom)
        }
        container.removeChild(cloned)
    }

    return { maxHeight, paddingTop, paddingBottom }
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export interface TransitionClassNamesOptions extends TransitionTimeoutOptions {
    classNames: FunctionMaybe<CSSTransitionClassNames>
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export function getTransitionClassNames({ classNames: cs, ...timeoutOptions }: TransitionClassNamesOptions): Readonly<Required<CSSTransitionClassNamesObject>> {
    const classNames = $$(cs)
    const timeout = getTransitionTimeout(timeoutOptions)

    if (typeof classNames === "string") {
        const { appear, enter, exit } = timeout
        return {
            appear: appear ? `${classNames}--appear` : "",
            appearActive: appear ? `${classNames}--appear-active` : "",
            appearDone: "",
            enter: enter ? `${classNames}--enter` : "",
            enterActive: enter ? `${classNames}--enter-active` : "",
            enterDone: "",
            exit: exit ? `${classNames}--exit` : "",
            exitActive: exit ? `${classNames}--exit-active` : "",
            exitDone: "",
        }
    }

    const {
        enter = "",
        enterActive = "",
        enterDone = "",
        exit = "",
        exitActive = "",
        exitDone = "",
        appear = (timeout.appear && enter) || "",
        appearActive = (timeout.appear && enterActive) || "",
        appearDone = (timeout.appear && enterDone) || "",
    } = classNames

    return {
        appear,
        appearActive,
        appearDone,
        enter,
        enterActive,
        enterDone,
        exit,
        exitActive,
        exitDone,
    }
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
interface IsWithinViewportOptions {
    fixedElement: HTMLElement
    fixedToElement: HTMLElement
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export function isWithinViewport({
    fixedElement,
    fixedToElement,
}: IsWithinViewportOptions): boolean {
    const fixedElementRect = fixedElement.getBoundingClientRect()
    const fixedToElementRect = fixedToElement.getBoundingClientRect()
    const vh = getViewportSize("height")
    const vw = getViewportSize("width")
    const top = Math.min(fixedElementRect.top, fixedToElementRect.top)
    const right = Math.max(fixedElementRect.right, fixedToElementRect.right)
    const bottom = Math.max(fixedElementRect.bottom, fixedToElementRect.bottom)
    const left = Math.min(fixedElementRect.left, fixedToElementRect.left)

    return bottom >= 0 && top <= vh && right >= 0 && left <= vw
}
