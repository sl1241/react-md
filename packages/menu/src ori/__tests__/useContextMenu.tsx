import { Configuration } from "@react-md/layout";
import {
  ContentCopySVGIcon,
  ContentCutSVGIcon,
  ContentPasteSVGIcon,
} from "@react-md/material-icons";
import { BELOW_CENTER_ANCHOR } from "@react-md/utils";
import type { RenderResult } from "@testing-library/react";
import { fireEvent, render as baseRender } from "@testing-library/react";
import type { FC, MouseEvent, ReactElement } from "react";

import { Menu } from "../Menu";
import { MenuItem } from "../MenuItem";
import type { ContextMenuHookOptions } from "../useContextMenu";
import { useContextMenu } from "../useContextMenu";

const Wrapper: FC = ({ children }) => (
  <Configuration disableRipple>{children}</Configuration>
);

function render(ui: ReactElement): RenderResult {
  return baseRender(ui, { wrapper: Wrapper });
}

type TestProps = ContextMenuHookOptions;

function Test(props: TestProps): ReactElement {
  const { menuRef, menuProps, onContextMenu } = useContextMenu(props);
  return (
    <>
      <div data-testid="container" onContextMenu={onContextMenu} />
      <Menu {...menuProps} ref={menuRef} timeout={0}>
        <MenuItem leftAddon={<ContentCutSVGIcon />}>Cut</MenuItem>
        <MenuItem leftAddon={<ContentCopySVGIcon />}>Copy</MenuItem>
        <MenuItem leftAddon={<ContentPasteSVGIcon />}>Paste</MenuItem>
      </Menu>
    </>
  );
}

describe("useContextMenu", () => {
  it("should work correctly", () => {
    const { getByTestId, getByRole } = render(<Test />);
    const container = getByTestId("container");

    expect(() => getByRole("menu")).toThrow();

    fireEvent.contextMenu(container, {
      clientX: 100,
      clientY: 50,
      button: 1,
      buttons: 1,
    });
    expect(document.body).toMatchSnapshot();
    const menu = getByRole("menu", { name: "Context Menu" });
    expect(menu).toBeInTheDocument();
  });

  it("should not display if a custom onContextMenu function calls event.stopPropagation()", () => {
    const onContextMenu = jest.fn((event: MouseEvent) => {
      event.stopPropagation();
    });

    const { getByTestId, getByRole } = render(
      <Test onContextMenu={onContextMenu} />
    );

    fireEvent.contextMenu(getByTestId("container"));
    expect(onContextMenu).toBeCalledTimes(1);
    expect(() => getByRole("menu")).toThrow();
  });

  it("should allow the anchor, baseId, and menuLabel to be configured", () => {
    const { getByTestId, getByRole } = render(
      <Test
        anchor={BELOW_CENTER_ANCHOR}
        baseId="custom-context-menu"
        menuLabel="Custom Label"
      />
    );
    const container = getByTestId("container");

    expect(() => getByRole("menu")).toThrow();

    fireEvent.contextMenu(container);
    expect(document.body).toMatchSnapshot();
    const menu = getByRole("menu", { name: "Custom Label" });
    expect(menu).toBeInTheDocument();
  });
});
