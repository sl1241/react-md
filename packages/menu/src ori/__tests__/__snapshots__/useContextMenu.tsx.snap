// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContextMenu should allow the anchor, baseId, and menuLabel to be configured 1`] = `
<body
  class="rmd-utils--mouse"
  data-rmd-noscroll=""
  style="overflow: hidden;"
>
  <div>
    <div
      data-testid="container"
    />
  </div>
  <div
    aria-label="Custom Label"
    class="rmd-menu rmd-menu--elevated"
    id="custom-context-menu-menu"
    role="menu"
    style="left: 16px; top: 0px; position: fixed; transform-origin: 50% 0;"
    tabindex="-1"
  >
    <ul
      class="rmd-list"
      role="none"
    >
      <li
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="rmd-icon rmd-icon--svg rmd-icon--before rmd-list-item__addon rmd-list-item__addon--before"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.64 7.64c.23-.5.36-1.05.36-1.64 0-2.21-1.79-4-4-4S2 3.79 2 6s1.79 4 4 4c.59 0 1.14-.13 1.64-.36L10 12l-2.36 2.36C7.14 14.13 6.59 14 6 14c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4c0-.59-.13-1.14-.36-1.64L12 14l7 7h3v-1L9.64 7.64zM6 8c-1.1 0-2-.89-2-2s.9-2 2-2 2 .89 2 2-.9 2-2 2zm0 12c-1.1 0-2-.89-2-2s.9-2 2-2 2 .89 2 2-.9 2-2 2zm6-7.5c-.28 0-.5-.22-.5-.5s.22-.5.5-.5.5.22.5.5-.22.5-.5.5zM19 3l-6 6 2 2 7-7V3z"
          />
        </svg>
        <span
          class="rmd-list-item__text"
        >
          Cut
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="rmd-icon rmd-icon--svg rmd-icon--before rmd-list-item__addon rmd-list-item__addon--before"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
        <span
          class="rmd-list-item__text"
        >
          Copy
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="rmd-icon rmd-icon--svg rmd-icon--before rmd-list-item__addon rmd-list-item__addon--before"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 2h-4.18C14.4.84 13.3 0 12 0c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm7 18H5V4h2v3h10V4h2v16z"
          />
        </svg>
        <span
          class="rmd-list-item__text"
        >
          Paste
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`useContextMenu should work correctly 1`] = `
<body
  class="rmd-utils--mouse"
  data-rmd-noscroll=""
  style="overflow: hidden;"
>
  <div>
    <div
      data-testid="container"
    />
  </div>
  <div
    aria-label="Context Menu"
    class="rmd-menu rmd-menu--elevated"
    id="context-menu-menu"
    role="menu"
    style="left: 100px; top: 50px; position: fixed; transform-origin: 0 0;"
    tabindex="-1"
  >
    <ul
      class="rmd-list"
      role="none"
    >
      <li
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="rmd-icon rmd-icon--svg rmd-icon--before rmd-list-item__addon rmd-list-item__addon--before"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.64 7.64c.23-.5.36-1.05.36-1.64 0-2.21-1.79-4-4-4S2 3.79 2 6s1.79 4 4 4c.59 0 1.14-.13 1.64-.36L10 12l-2.36 2.36C7.14 14.13 6.59 14 6 14c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4c0-.59-.13-1.14-.36-1.64L12 14l7 7h3v-1L9.64 7.64zM6 8c-1.1 0-2-.89-2-2s.9-2 2-2 2 .89 2 2-.9 2-2 2zm0 12c-1.1 0-2-.89-2-2s.9-2 2-2 2 .89 2 2-.9 2-2 2zm6-7.5c-.28 0-.5-.22-.5-.5s.22-.5.5-.5.5.22.5.5-.22.5-.5.5zM19 3l-6 6 2 2 7-7V3z"
          />
        </svg>
        <span
          class="rmd-list-item__text"
        >
          Cut
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="rmd-icon rmd-icon--svg rmd-icon--before rmd-list-item__addon rmd-list-item__addon--before"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
        <span
          class="rmd-list-item__text"
        >
          Copy
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="rmd-icon rmd-icon--svg rmd-icon--before rmd-list-item__addon rmd-list-item__addon--before"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 2h-4.18C14.4.84 13.3 0 12 0c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm7 18H5V4h2v3h10V4h2v16z"
          />
        </svg>
        <span
          class="rmd-list-item__text"
        >
          Paste
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
    </ul>
  </div>
</body>
`;
